[Store] 매장 폐업 기능 구현

iop0217 opened the pull request:
### Description
- 매장 폐업 기능을 구현했습니다.
- 매장 삭제가 아닌 storeStatus를 CLOSED로 변경하는 방식으로 폐업을 처리합니다.
- 본인 소유 매장만 폐업할 수 있도록 권한 검증 로직을 추가했습니다.
- 폐업된 매장은 매장 전체 조회 및 단건 조회 시 결과에 노출되지 않도록 했습니다.
- 폐업 성공 시 204 No Content를 반환합니다.
- store 폐업 처리 로직을 delete() 메서드로 통합했습니다.

---

### Changes
- `DELETE /api/stores/{id}` 매장 폐업 API 구현
- Stores 엔티티에 close() 메서드 추가
- StoreService 및 StoreController에 폐업 처리 로직 추가
- 매장 전체 조회 및 단건 조회 시 storeStatus = LIVE 조건 추가
- StoreRepository에 상태 기반 조회 메서드 추가
- 폐업 성공/실패 테스트 코드 작성
- Postman을 통해 정상 응답 테스트 완료

---

### Screenshots

#### ✅ 204 No Content
![204](https://github.com/user-attachments/assets/64858e86-bb68-4e0e-b9f8-11fcb2e3674b)

#### ❌ 401 Unauthorized
![401](https://github.com/user-attachments/assets/c3c3919e-c142-486f-bf6f-4321c99ae327)

#### ❌ 403 Forbidden
![403](https://github.com/user-attachments/assets/a0d68abc-e9dd-4531-a1f3-8e6f91846ede)

#### ❌ 404 Not Found
![404](https://github.com/user-attachments/assets/6f2d1a78-e6c9-438f-9ab4-dbd093c63cf8)

iop0217 committed:
feature[store]: 매장 폐업 기능 구현

- 매장 폐업 시 storeStatus를 CLOSED로 변경
- 본인 소유 매장만 폐업 가능하도록 권한 검증

iop0217 committed:
test[store]: 매장 폐업 테스트 코드 추가

- 매장 폐업 성공 테스트 추가
- 매장 소유자 불일치로 인한 폐업 실패 테스트 추가
- 매장 미존재로 인한 폐업 실패 테스트 추가

iop0217 committed:
Merge branch 'dev' into feature/suhyun-stores

iop0217 committed:
fix[store]: 폐업 매장이 조회되는 문제 수정

- 폐업된 매장이 조회되지 않도록 조회 방식 수정
- 전체 조회 및 단건 조회 메서드 일부 변경
- StoreRepository에 조회 메서드 추가

anyubinn commented on src/main/java/com/delivery/igo/igo_delivery/api/store/service/StoreServiceImpl.java:
```diff
@@ -125,4 +127,19 @@ public StoreUpdateResponseDto updateStore(Long storeId, Long authUserId, StoreUp
+            throw new GlobalException(ErrorCode.STORE_OWNER_MISMATCH);
+        }
+
+        store.close();
```
store 엔티티에 있는 delete와 겹치는 것 같아요! 두 메서드 중 하나를 제거하고 삭제일과 매장 상태를 같이 업데이트하는 것이 좋을 것 같아요

iop0217 replied:
delete는 초반 세팅할 때부터 있던 메서드라 제가 건드려도 될지 모르겠어서 놔뒀었는데 너무 좋은 것 같아요
수정했어요

iop0217 committed:
refactor[store]: 매장 폐업 로직 통합 수정

- store.close() 메서드 삭제
- store.delete() 메서드에서 삭제일 기록 및 상태 변경(CLOSED) 처리
- 폐업 로직에서 delete() 호출

iop0217 committed:
Merge branch 'dev' into feature/suhyun-stores
